### PR TITLE
Add StatsCommand to AddressBookParser for command parsing

### PR DESCRIPTION
**Pull request by AlekKwek18**
- Import StatsCommand in AddressBookParser to handle the "stats" command.
- Add a new case in the parser’s switch statement to recognize and execute StatsCommand.
- Ensure that users can access the statistics functionality through the "stats" command in the UI.

Resolves issue #238

Screenshot 2024-11-06 at 12 17 45 AM

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 1]
```diff
@@ -93,6 +94,8 @@ public Command parseCommand(String userInput) throws ParseException {
         case UnmatchCommand.COMMAND_WORD:
             return new UnmatchCommandParser().parse(arguments);
 
+        case StatsCommand.COMMAND_WORD:
```
Comment (KengHian) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:97]: Kindly update the UG as well.

**Review by KengHian — APPROVED** [review]
Overall LGTM, just have some questions.

File: src/main/java/seedu/address/logic/commands/StatsCommand.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        int totalContacts = model.getFilteredPersonList().size();
+        int totalJobs = model.getFilteredJobList().size();
+        int totalCompanies = model.getFilteredCompanyList().size();
+
+        long matchedContacts = model.getFilteredPersonList().stream()
+                .filter(Person::isMatchPresent)
+                .count();
+        long unmatchedContacts = totalContacts - matchedContacts;
+
+        String resultMessage = String.format("Total Contacts: %d\nTotal Jobs: %d\nTotal Companies: "
```
[the remark this answers is not in the document]
Comment (Green-Tea-123) [reply src/main/java/seedu/address/logic/commands/StatsCommand.java:28]: agree.

**Review by Green-Tea-123 — APPROVED** [review]
Overall LGTM.

**Pull request merged**